<launch>
  <!-- The planning and execution components of MoveIt! configured to run -->
  <!-- using the ROS-Industrial interface. -->
 
  <!-- Non-standard joint names:
       - Create a file [robot_moveit_config]/config/joint_names.yaml
           controller_joint_names: [joint_1, joint_2, ... joint_N] 
       - Update with joint names for your robot (in order expected by rbt controller)
       - and uncomment the following line: -->
  <rosparam command="load" file="$(find hc10_ros)/config/joint_names.yaml"/>
 
  <!-- the "sim" argument controls whether we connect to a Simulated or Real robot -->
  <!--  - if sim=false, a robot_ip argument is required -->
  <arg name="sim" default="true" />
  <!--Uncomment the lines below if you want to fixate the robot_ip address and its controller-->
  <!--<arg name="robot_ip" default="192.168.0.113" unless="$(arg sim)" />
  <arg name="controller" default="yrc1000" unless="$(arg sim)" /> -->
 
  <!-- load the robot_description parameter before launching ROS-I nodes -->
  <include file="$(find hc10_ros)/launch/planning_context.launch" >
    <arg name="load_robot_description" value="true" />
  </include>

  <!-- run the robot simulator and action interface nodes -->
  <group if="$(arg sim)">
    <include file="$(find industrial_robot_simulator)/launch/robot_interface_simulator.launch" />
  </group>

  <!-- run the "real robot" interface nodes -->
  <!--   - this typically includes: robot_state, motion_interface, and joint_trajectory_action nodes -->
  <!--   - replace these calls with appropriate robot-specific calls or launch files -->
  <group unless="$(arg sim)">
    <include file="$(find motoman_hc10_support)/launch/robot_interface_streaming_hc10.launch" >
      <arg name="robot_ip" value="$(arg robot_ip)"/>
      <arg name="controller" value="$(arg controller)"/>
    </include>
    
  </group>

  <!-- publish the robot state (tf transforms) -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

  <include file="$(find hc10_ros)/launch/move_group.launch">    <arg name="publish_monitored_planning_scene" value="true" />
  </include>

  <include file="$(find hc10_ros)/launch/moveit_rviz.launch">
    <arg name="config" value="true"/>
  </include>

  <!-- add interferences using python script -->
  <!-- <node name="add_interferences" pkg="hc10_ros" type="add_interferences.py" /> -->
</launch>
